src/Chapter3.hs

```diff
@@ -480,6 +531,36 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
 
+data Castle   = NoCastle | Castle String      deriving (Show)
+data Wall     = NoWall   | HasWall            deriving (Show)
+data ChOrLib  = Church   | Library            deriving (Show)
+data House    = House { housePeople :: Int }  deriving (Show)
+
+data City = City
+  { cityCastle  :: Castle
+  , cityWall    :: Wall
+  , cityChOrLib :: ChOrLib
+  , cityHouse   :: [House]
+  } deriving (Show)
+
+buildCastle :: City -> String -> City
+buildCastle city name = city { cityCastle = Castle name }
+
+buildHouse :: City -> Int -> City
+buildHouse city people = city { cityHouse = House people : (cityHouse city) }
```

Comment:
This function takes `Int`, which means that it's possible to give any `Int` to it. But according to our task description, a house must contain between 1 and 4 people

Reply:
I understand that the number of people in a house is limited t o1, 2, 3, and 4. I revised the code and pushed.